### ISSUE TITLE
Anaconda is spamming process monitor with tasklist

### ISSUE THREAD

**Issue by evandrocoan** [DamnWidget/anaconda#695]
> If you are not running ST3 there shouldn't be any instance of jsonserver running in your machine, in fact, the jsonserver will kill himself if the plugin_host application is not running. That is checked using the windows command line tool tasklist by the jsonserver each five seaconds.

https://github.com/DamnWidget/anaconda/issues/67#issuecomment-24433751

This is spamming my console on process hacker:

![image](https://user-images.githubusercontent.com/5332158/29775611-226426ac-8bdc-11e7-82f0-0de65bf73dbc.png)

Is this nice to do with the users monitors? Perhaps this time should be increased or done by any other means than polling `tasklist` all the time.

**Comment by DamnWidget** [comment]
Why is the tasklist process still running? Can you try to run tasklist from a CMD.exe or Powershell to see if it hangs?

**Comment by evandrocoan** [comment]
I do not understand `is the tasklist process still running`, I had Sublime Text running for several days, and looked over my console and saw anaconda spamming these `tasklist` processes. 

The tasklist process is not hanged, it is spamming every few seconds:

![back](https://user-images.githubusercontent.com/5332158/29779566-4d497b6c-8be9-11e7-8b59-d844f859408b.gif)

**Comment by evandrocoan** [comment]
I opened the file:
```python
File: Anaconda/anaconda_server/jsonserver.py
194:         self.delta = 300
```
And set the delta from 5 to 300 in hope it will stop spamming on the future.

**Comment by DamnWidget** [comment]
`tasklist.exe` is a terminal/console application for windows that show you a list of your running processes. The process is not a service or daemon so is like if you use `cd` or `dir` it shouldn't be shown in your active processes as it will just return the list of processes and die.

Anaconda does not spam anything, it just calls that windows command to know if ST3 is still running in your system or not, I asked you if you could run tasklist from your terminal to see how it behaves and you answered with more images that are useless, if you are not willing to follow simple instructions and give me results I am afraid that I can not give you any support.

**Comment by evandrocoan** [comment]
I answered your question:

>  the tasklist process is not hanged

Of course I run it form the command line. It just outputs all my running processes. So it is again not hanged.

**Comment by evandrocoan** [comment]
>  it just calls that windows command to know if ST3 is still running in your system or not

That is the problem, calling it every few seconds to know whether Sublime Text is running or not. My processo monitor shows all system process running, therefore all Anaconda calls to `tasklist` will show up. I never had any program which did such thing. 

After this change:
```python
File: Anaconda/anaconda_server/jsonserver.py
194:         self.delta = 300
```
The issue seems to be fixed.

**Comment by evandrocoan** [comment]
> I asked you if you could run tasklist from your terminal to see how it behaves and you answered with more images that are useless, if you are not willing to follow simple instructions and give me results I am afraid that I can not give you any support.

Please give clear instructions if you want me to understand. Saying:

> Why is the tasklist process still running? Can you try to run tasklist from a CMD.exe or Powershell to see if it hangs?

Is everything but not clear on about what you want to know.

**Comment by DamnWidget** [comment]
You should not see any tasklist process as the process should return as soon as it runs (just as when you run it in the terminal). First of all, lets determine things in here because I am not a Windows user and I have no access to any Windows machine right now. Which program are you using to see your running processes and what it means that the background is red?

If you run `tasklist` yourself, do you see a lot of other `tasklist` processes running as well?

**Comment by evandrocoan** [comment]
Thanks for the patience, and sorry for my uncleanness.

> You should not see any tasklist process as the process should return as soon as it runs (just as when you run it in the terminal).

Use Process Hacker (http://processhacker.sourceforge.net/) to manage and monitor my processes. I configured it to show all closed process for at least `600.000` seconds.

> what it means that the background is red?

So, when I process exists, it background on Process Hacker is show in the red color. I using this to know what processes were running on my machine some time in the past. I use this to spot processes which ran but should not, or are just eating my resources/CPU for no reason.

Now, if Anaconda server launches/spawn a new `tasklist` process each 3 or 5 seconds, it will leave a huge trail of dead processes on my list which remember the dead process for `600.000` seconds. The animation above shows the `tasklist` process being launch (green color background) and when they die, their background became red (red color background). So they are pilling up dead process every few seconds.

> if you run tasklist yourself, do you see a lot of other tasklist processes running as well?

It just shows the list of running processes:
```
$ tasklist

Image Name                     PID Session Name        Session#    Mem Usage
========================= ======== ================ =========== ============
System Idle Process              0 Services                   0          8 K
System                           4 Services                   0        980 K
...
```

**Comment by DamnWidget** [comment]
But that is just a regular process operation, the process does not exists anymore in your system and is not using any resources, you are just concerned because you have a task manager that caches the execution of processes for 10 minutes.

Anaconda fires up a `tasklist` process every 5 seconds precisely to don't usu any resources from your system in case that the ST3 editor is closed, is 5 seconds because we want to stop the Python server as soon as we can to don't waste resources in your machine.

You have three options:
1) Add tasklist to any kind of ignore filter in your tasks monitor application 
2) Ignore (mentally) all the tasklist garbage that you see in red (and I repeat those are not real processes in your system as the process was long gone)
3) You edit the `jsonserver.py` file in every release yourself and add whatever delay you consider for the processes monitor 

I am afraid that this is not an issue for any other user in the world so I am closing this.
